snapcraft/file_utils.py

```diff
@@ -343,23 +343,24 @@ def get_tool_path(command_name: str) -> str:
     :return: Path to command
     :rtype: str
     """
+    snapcraft_snap_path = os.path.join(os.sep, "snap", "snapcraft", "current")
+
     if common.is_snap():
         command_path = _command_path_in_root(os.getenv("SNAP"), command_name)
-    elif common.is_docker_instance():
-        command_path = _command_path_in_root(
-            os.path.join(os.sep, "snap", "snapcraft", "current"), command_name
-        )
+    elif common.is_docker_instance() and os.path.exists(snapcraft_snap_path):
```

Comment:
`is_snap()` is true if `$SNAP_NAME == "snapcraft"`. Due to [this file](https://github.com/snapcore/snapcraft/blob/master/docker/bin/snapcraft-wrapper) it will also be true in the snap-based Docker images, which means I don't think this `elif` will ever be true.

Reply:
Good catch, removed the condition as discussed during our quick call about it.